### ISSUE TITLE
Lerna publish fails could not find package.json

### ISSUE THREAD

**Issue by chemscn** [lerna/lerna#3279]
Currently creating vue monorepo component library with vue2, vite and lerna.

Everytime I try to publish using `lerna publish`, I get the following error:

![image](https://user-images.githubusercontent.com/77703084/183103948-3e1617be-e9df-482a-a793-979f8aa1bdb2.png)

Now it is true the vite build is not creating the package.json in the dist folder.

Is the package.json required to publish??

**Comment by ghiscoding** [comment]
When using Lerna, you must have a `package.json` in the project root (where you add Lerna as devDependency) and also have `package.json` files in each of your monorepo packages (these will be needed to publish each of your packages). You can check this YouTube video [How to Use Lerna](https://www.youtube.com/watch?v=p6qoJ4apCjA) which got me started when I first looked at using it

**Comment by msobiecki** [comment]
> When using Lerna, you must have a `package.json` in the project root (where you add Lerna as devDependency) and also have `package.json` files in each of your monorepo packages (these will be needed to publish each of your packages). You can check this YouTube video [How to Use Lerna](https://www.youtube.com/watch?v=p6qoJ4apCjA) which got me started when I first looked at using it

![image](https://user-images.githubusercontent.com/9337412/189664336-2047d8d9-ff4c-4f05-baa0-bc4667be9e7e.png)

Should Lerna really catch the nasted package.json, even in folders like dist / build / .next that are added to gitignore?

![image](https://user-images.githubusercontent.com/9337412/189664954-2593dcc6-c525-4349-a145-9fe4ef7c0b6a.png)

It seems to me that it should only catch those from the level specified in lerna.json, not deeper?

**Comment by msobiecki** [comment]
> you have `useWorkspaces` set to true, so I assume that it will follow the `workspaces` that is defined in your `package.json` and completely ignore the `packages` set in `lerna.json`

Yes that's right, `packages` in lerna.json is some debt after the old implementation, even the documentation says it (https://lerna.js.org/docs/api-reference/configuration#useworkspaces--packages).

Properly configuring workspaces in pnpm-workspace.yaml solved my problem. Thanks for pointing to the solution @ghiscoding 🥇
